iocore/net/P_SSLConfig.h

```diff
@@ -134,8 +346,8 @@ struct SSLConfigParams : public ConfigInfo {
   // The first level maps from CA bundle file&path to next level map;
   // The second level maps from cert&key to actual SSL_CTX;
   // The second level map owns the client SSL_CTX objects and is responsible for cleaning them up
-  using CTX_MAP = std::unordered_map<std::string, shared_SSL_CTX>;
-  mutable std::unordered_map<std::string, CTX_MAP> top_level_ctx_map;
+  using CTX_MAP = std::unordered_map<SSLUtilsTwoCStrKey, shared_SSL_CTX, SSLUtilsImpl::TwoCStrKeyHash>;
+  mutable std::unordered_map<SSLUtilsTwoCStrKey, CTX_MAP, SSLUtilsImpl::TwoCStrKeyHash> top_level_ctx_map;
```

Comment:
A vulnerability with these maps is that the emplace() member function will not work right (we currently don't use this).  Instances of SSLUtilsTwoCStrKey are either transient or empty, unless they were created with a copy/move constructor, or assigned.